src/workerd/server/server.c++

```diff
@@ -703,13 +705,16 @@ class Server::ExternalHttpService final: public Service, private kj::TaskSet::Er
         kj::HttpService::Response& response) override {
       TRACE_EVENT("workerd", "ExternalHttpServer::request()");
       KJ_REQUIRE(wrappedResponse == kj::none, "object should only receive one request");
-      wrappedResponse = response;
+      if (parent.rewriter->needsRewriteResponse()) {
+        wrappedResponse = response;
+      }
       if (parent.rewriter->needsRewriteRequest()) {
         auto rewrite = parent.rewriter->rewriteOutgoingRequest(url, headers, metadata.cfBlobJson);
-        return parent.serviceAdapter->request(method, url, *rewrite.headers, requestBody, *this)
-            .attach(kj::mv(rewrite));
+        co_return co_await parent.serviceAdapter->request(method, url, *rewrite.headers,
+            requestBody, parent.rewriter->needsRewriteResponse() ? *this : response);
       } else {
-        return parent.serviceAdapter->request(method, url, headers, requestBody, *this);
+        co_return co_await parent.serviceAdapter->request(method, url, headers, requestBody,
```

Comment:
There are a bunch of changes in this file which don't seem related to containers. Some of them appear to be converting methods to coroutines for no particular reason.

If these are not strictly needed for the containers change, please revert these changes. If they are needed, please explain why.

If these are just separate refactorings you wanted to do, please make a separate PR for them. Please note, though, replacing `return` with `co_return co_await` hurts performance and may have unintended side effects; we should not do it unless it actually has some other benefit to the code.